### PR TITLE
♻️ HOCS-4383: Remove json dependency and reduce the amount of deadline calculations

### PR DESCRIPTION
**Pull request by nickmcmahon01**
Remove needless json dependency and reduce the amount of deadline calculations performed

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java [hunk 1]
```diff
@@ -288,33 +286,19 @@ void updateContributions(Set<StageWithCaseData> stage) {
     StageWithCaseData getUnassignedAndActiveStageByTeamUUID(UUID teamUUID, UUID userUUID) {
         log.debug("Getting unassigned cases for user: {} in team {}", userUUID, teamUUID);
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java:287]: Method returns a single but refers to a plural. Does the name of this method make sense at all?

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java [hunk 2]
```diff
@@ -288,33 +286,19 @@ void updateContributions(Set<StageWithCaseData> stage) {
...
-            daysElapsedCalculator.updateDaysElapsed(stage.getData(), stage.getCaseDataType());
-        }
+        unassignedStages.forEach(stage -> stagePriorityCalculator.updatePriority(stage.getData(), stage.getCaseDataType()));
 
-        double prevSystemCalculatedPriority = 0;
-        StageWithCaseData nextAvailableStage = unassignedStages.stream().findFirst().get();
-        for (StageWithCaseData stage : unassignedStages) {
-            JSONObject caseData = new JSONObject(stage.getData());
-            double systemCalculatedPriority = caseData.getDouble("systemCalculatedPriority");
+        var nextAvailableStage = unassignedStages.stream().max(Comparator.comparingDouble(stage -> Double.parseDouble(stage.getData().getOrDefault("systemCalculatedPriority", "0"))));
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java:292]: Magic number/ string, what does 0 actually mean?

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java [hunk 3]
```diff
@@ -288,33 +286,19 @@ void updateContributions(Set<StageWithCaseData> stage) {
...
 
-            if (systemCalculatedPriority > prevSystemCalculatedPriority) {
-                prevSystemCalculatedPriority = systemCalculatedPriority;
-                nextAvailableStage = stage;
-            }
+        if(nextAvailableStage.isPresent()) {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java:294]: Fancy adding a line after if, pretty please?…

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java [hunk 4]
```diff
@@ -288,33 +286,19 @@ void updateContributions(Set<StageWithCaseData> stage) {
...
+            StageWithCaseData stage = nextAvailableStage.get();
+            updateStageUser(stage.getCaseUUID(), stage.getUuid(), userUUID);
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java:296]: Nice catch this to reduce complexity and calculations…

File: build.gradle [hunk 5]
```diff
@@ -56,7 +56,6 @@ dependencies {
     implementation("org.apache.camel:camel-spring-boot-starter:${camelVersion}")
     implementation("org.apache.camel:camel-aws:${camelVersion}")
     implementation('com.amazonaws:aws-java-sdk:1.12.131')
-    implementation 'org.json:json:20211205'
```
Comment (LiamMacP) [inline build.gradle:59]: Was this used anywhere?

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java (hunk 2 above)
Comment (nickmcmahon01) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java:292]: 'priority'

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java (hunk 4 above)
Comment (nickmcmahon01) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java:296]: again, we were calculating x hundred deadlines, then filtering to first().

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java (hunk 1 above)
Comment (nickmcmahon01) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java:287]: we're not even fetching cases.

**Comment by nickmcmahon01** [comment]
I'm not happy with the lack of test coverage for this method either, I'll add some.

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java (hunk 2 above)
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java:292]: Change to a const, `NON_PRIORITY` or something...

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java (hunk 3 above)
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java:294]: Flip the if and return null early?

**Pull request closed without merging**